plugins/kubernetes/app/assets/javascripts/kubernetes/controllers/kubernetes_create_role_ctrl.js

```diff
@@ -0,0 +1,40 @@
+samson.controller('KubernetesCreateRoleCtrl', function($scope, $state, $stateParams, kubernetesService, kubernetesRoleFactory, notificationService, $timeout) {
+
+  $scope.submit = function(e) {
+    e.preventDefault(); //prevents default form submission
+
+    kubernetesService.createRole($stateParams.project_id, $scope.role).then(
+      function() {
+        // Postpones execution of this instruction until the current digest cycle is finished.
+        // This was needed to keep the flash message on the page, otherwise it would disappear
+        // instantly due to the URL state change that follows.
+        $timeout(function() {
+          notificationService.success('Role ' + $scope.role.name + ' has been created successfully.');
+        });
```

Comment:
This looks odd. Can you trigger the state change first and then trigger the flash msg?